core/lib/globals_shared.h

```diff
@@ -2017,4 +2017,30 @@ typedef struct _priv_mcontext_t {
 #include "mcxtx.h"
 } priv_mcontext_t;
 
+/* DR_API EXPORT BEGIN */
+/**
+ * Encodings of an application's argument.
+ */
+typedef enum {
+    APP_ARG_ASCII,
```

Comment:
I think this can be removed.  I think any modern UNIX environment is going to encode as UTF-8 or ISO-8859* or somesuch.  I think we want to be agnostic of which: we just need to know whether we can treat it as a null-terminated char* or not (i.e., it has no encoding which contains a zero byte).  Not sure if there's a standard label for that: `APP_ARG_CSTR_COMPAT`?  So I'm suggesting just having the cstring-compat and UTF-16.

Hmm: so dr_app_arg_as_utf8() is not as accurately named: maybe dr_app_arg_as_cstring()??

Reply:
Done.